### PR TITLE
Optimize the C++ code after running code quality check (Warnings) 

### PR DESCRIPTION
**Pull request by jeevithiesh**

### CODE REVIEW

File: goa/Classes/alphamon/Alphamon.cpp [hunk 1]
```diff
@@ -239,8 +239,8 @@ void Alphamon::alphamonLegAnimation(std::string animationName, bool loop) {
 	for (auto item = children.rbegin(); item != children.rend(); ++item) {
 		Node * monsterItem = *item;
 		std::string str = monsterItem->getName().c_str();
-		if (str.find("skate") == 0) {
-			auto legTimeline = CSLoader::createTimeline(CCString::createWithFormat("leg_ani/%s.csb", str.c_str())->getCString());
+        if (str.compare(0, 5, "skate")){
```
Comment (ankitk20) [inline goa/Classes/alphamon/Alphamon.cpp]: There has to be a space here

File: goa/Classes/mini_games/AlphaPhonics.cpp [hunk 2]
```diff
@@ -177,7 +177,7 @@ void AlphaPhonics::OptionListner(Sprite *option) {
 	_eventDispatcher->addEventListenerWithSceneGraphPriority(listener, option);
 }
 
-string AlphaPhonics::StringandIntConcat(string data,int number) {
+string AlphaPhonics::StringandIntConcat(const string data,int number) {
```
Comment (ankitk20) [inline goa/Classes/mini_games/AlphaPhonics.cpp]: Space after comma

File: goa/Classes/mini_games/BasicMultiplication.cpp [hunk 3]
```diff
@@ -110,7 +110,7 @@ void BasicMultiplication::topBoardSetting() {
 	_questionValue = getRandomValueRange(1, 10, 5);
 }
 
-void BasicMultiplication::gridGrayAndListnerController(string gridName,int row , int column ) {
+void BasicMultiplication::gridGrayAndListnerController(const string gridName,int row , int column ) {
```
Comment (ankitk20) [inline goa/Classes/mini_games/BasicMultiplication.cpp]: Space after comma

File: goa/Classes/mini_games/ChocolateFactory.cpp [hunk 4]
```diff
@@ -119,7 +119,7 @@ void ChocolateFactory::onEnterTransitionDidFinish()
 		for (int j=0; j<sprite->getChildren().size(); j++)
 		{ 
 			 std::string str = sprite->getChildren().at(j)->getName();
-			 if(str.find("milkcarton")==0)
+			 if(str.compare(0, 10, "milkcarton")==0)
```
Comment (ankitk20) [inline goa/Classes/mini_games/ChocolateFactory.cpp]: Space before and after ==

File: goa/Classes/util/Speaker.cpp [hunk 5]
```diff
@@ -31,7 +31,7 @@ Speaker::~Speaker() {
  position - Vec type,to set position of widget ....
  selectionMode - select choice option enable if true or default false
  */
-Speaker*  Speaker::createSpeaker(string word,Vec2 position,bool selectionMode) {
+Speaker*  Speaker::createSpeaker(const string word, Vec2 position,bool selectionMode) {
```
Comment (ankitk20) [inline goa/Classes/util/Speaker.cpp]: Space after comma

**Review by ankitk20 — CHANGES_REQUESTED** [review]

File: goa/Classes/alphamon/Alphamon.cpp (hunk 1 above)
Comment (jeevithiesh) [reply goa/Classes/alphamon/Alphamon.cpp]: I have added a space.

File: goa/Classes/mini_games/AlphaPhonics.cpp (hunk 2 above)
Comment (jeevithiesh) [reply goa/Classes/mini_games/AlphaPhonics.cpp]: I have added a space.

File: goa/Classes/mini_games/ChocolateFactory.cpp (hunk 4 above)
Comment (jeevithiesh) [reply goa/Classes/mini_games/ChocolateFactory.cpp]: I have added a space.

File: goa/Classes/mini_games/BasicMultiplication.cpp (hunk 3 above)
Comment (jeevithiesh) [reply goa/Classes/mini_games/BasicMultiplication.cpp]: I have added a space.

File: goa/Classes/util/Speaker.cpp (hunk 5 above)
Comment (jeevithiesh) [reply goa/Classes/util/Speaker.cpp]: I have added a space.

**Pull request closed without merging**